Framework/Core/include/Framework/DataDescriptorMatcher.inc

```diff
@@ -80,23 +80,41 @@ std::ostream& operator<<(std::ostream& os, ValueHolder<V> const& holder)
 inline OriginValueMatcher::OriginValueMatcher(std::string const& s)
   : ValueHolder{ s }
 {
+  if (s.size() > header::gSizeDataOriginString) {
+    throw std::runtime_error("Header DataDescription too long");
+  }
 }
 
 inline OriginValueMatcher::OriginValueMatcher(ContextRef variableId)
   : ValueHolder{ variableId }
 {
 }
 
+template <std::size_t L>
+inline constexpr OriginValueMatcher::OriginValueMatcher(const char (&s)[L])
+  : ValueHolder{ header::DataOrigin(s).as<std::string>() }
+{
+}
+
 inline DescriptionValueMatcher::DescriptionValueMatcher(std::string const& s)
   : ValueHolder{ s }
 {
+  if (s.size() > header::gSizeDataDescriptionString) {
+    throw std::runtime_error("Header DataDescription too long");
+  }
 }
 
 inline DescriptionValueMatcher::DescriptionValueMatcher(ContextRef variableId)
   : ValueHolder{ variableId }
 {
 }
 
+template <std::size_t L>
+inline constexpr DescriptionValueMatcher::DescriptionValueMatcher(const char (&s)[L])
```

Comment:
constexpr already implies inline

Reply:
seems so, but is it a problem to have both?
When all other functions there are inline, it seems natural to me to put inline also to the constexpr ones, but I also don't care much.